src/chromatix/functional/lenses.py

```diff
@@ -65,6 +73,57 @@ def ff_lens(
     return optical_fft(field, f, n)
 
 
+def high_na_lens(
+    field: Field,
+    f: float,
+    n: float,
+    NA: float,
+    output_shape: Tuple[int, int],
```

Comment:
We don't always want to zoom / resample right? Should we have a default for output shape and dx to be None, and in this case we just use the shape of the incoming field?